wicket-core/src/main/java/org/apache/wicket/Component.java

```diff
@@ -2669,7 +2669,7 @@ public void internalRenderHead(final HtmlHeaderContainer container)
 					if (response.wasRendered(behavior) == false)
```

Comment:
The change itself looks good to me, but I'm wondering if the code is currently working as expected. We mark a pair of Component+Behavior as rendered, but the check if a behavior was already rendered only tests for the behavior itself.

@martin-g: You originally introduced this to fix https://issues.apache.org/jira/browse/WICKET-4570 many years ago. Could it be that you forgot to adjust the `wasRendered` condition? Imho the current code will never return `true` for `response.wasRendered(behavior)`.

Reply:
It looks like a bug indeed!
But it is strange that it was uncovered for so many years.

Reply:
Should we fix this? Or should we get rid of the whole conditional and always call `behavior.renderHead` as we seem to have been doing for 10 years without issues. All tests pass in both solutions.

Reply:
+1 to fix it

Reply:
@martin-g: I pushed https://github.com/apache/wicket/pull/515 that fixes this and incorporates the change suggested in this PR.